babel/languages.py

```diff
@@ -0,0 +1,61 @@
+# -- encoding: UTF-8 --
+from babel.core import get_global
+
+
+def get_official_languages(territory, regional=False, de_facto=False):
+    """
+    Get the official language(s) for the given territory.
+
+    The language codes, if any are known, are returned in order of descending popularity.
+
+    If the `regional` flag is set, then languages which are regionally official are also returned.
+
+    If the `de_facto` flag is set, then languages which are "de facto" official are also returned.
+
+    :param territory: Territory code
+    :type territory: str
+    :param regional: Whether to return regionally official languages too
+    :type regional: bool
+    :param de_facto: Whether to return de-facto official languages too
+    :type de_facto: bool
+    :return: Tuple of language codes
+    :rtype: tuple[str]
+    """
+
+    territory = str(territory).upper()
+    allowed_stati = set(("official",))
+    if regional:
+        allowed_stati.add("official_regional")
+    if de_facto:
+        allowed_stati.add("de_facto_official")
+
+    pairs = []
+    for language, info in get_global("territory_languages").get(territory, {}).items():
+        if info["official_status"] in allowed_stati:
+            pairs.append((info["population_percent"], language))
+    pairs.sort(reverse=True)
```

Comment:
I would be inclined to write:

```python
languages = get_global("territory_languages").get(territory, {})
pairs = [(info['population_percent'], language)
             for language, info in langs.iteritems()
                 if info['official_status'] in allowed_stati]
pairs.sort(reverse=True)
```

It should be a little faster and a bit more Pythonic to make more extensive use of comprehensions.